torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -616,6 +626,72 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
   return flattened;
 }
 
+void checkSplitSizes(
+    const std::vector<int64_t>& split_sizes,
+    const at::Tensor& tensor,
+    int group_size) {
+  if (split_sizes.size() == 0) {
+    if (tensor.size(0) % group_size != 0) {
+      throw std::runtime_error(
+          "Tensor's dim 0 does not divide equally across group size");
+    }
+  } else {
+    if (split_sizes.size() != group_size) {
+      throw std::runtime_error(
+          "Number of tensor splits not equal to group size");
+    }
+    int sum = std::accumulate(split_sizes.begin(), split_sizes.end(), 0);
+    if (sum != tensor.size(0)) {
+      throw std::runtime_error("Split sizes doesn't match total dim 0 size");
+    }
+  }
+}
+
+int64_t computeLengthsAndOffsets(
```

Comment:
This function and the function below are also duplication with the ones in `ProcessGroupMPI.cpp`?